src/devices/video/v9938.cpp

```diff
@@ -1138,43 +1138,72 @@ void v99x8_device::mode_graphic1(uint32_t *ln, int line)
 
 void v99x8_device::mode_graphic23(uint32_t *ln, int line)
 {
-	pen_t fg, bg, pen;
-	int nametbl_addr, patterntbl_addr, colourtbl_addr;
-	int pattern, x, xx, line2, name, charcode,
-	colour, colourmask, patternmask, xxx;
-
-	colourmask = ((m_cont_reg[3] & 0x7f) * 8) | 7;
-	patternmask = ((m_cont_reg[4] & 0x03) * 256) | 0xff;
-
-	nametbl_addr =  (m_cont_reg[2] << 10);
-	colourtbl_addr =  ((m_cont_reg[3] & 0x80) << 6) + (m_cont_reg[10] << 14);
-	patterntbl_addr = ((m_cont_reg[4] & 0x3c) << 11);
-
-	line2 = (line + m_cont_reg[23]) & 255;
-	name = (line2/8)*32;
-
-	pen = pen16(m_cont_reg[7] & 0x0f);
-	xxx = m_offset_x * 2;
-	while (xxx--) *ln++ = pen;
+	const int colourmask = ((m_cont_reg[3] & 0x7f) * 8) | 7;
+	const int patternmask = ((m_cont_reg[4] & 0x03) * 256) | 0xff;
+	const int nametbl_addr = m_cont_reg[2] << 10;
+	const int colourtbl_addr = ((m_cont_reg[3] & 0x80) << 6) + (m_cont_reg[10] << 14);
+	const int patterntbl_addr = ((m_cont_reg[4] & 0x3c) << 11);
+	const int scrolled_y = (line + m_cont_reg[23]) & 255;
+	const pen_t border_pen = pen16(m_cont_reg[7] & 0x0f);
+
+	int name = (scrolled_y / 8) * 32;
+
+	for (int x = m_offset_x * 2; x > 0; x--)
+		*ln++ = border_pen;
+
+	int scroll_to_mask = 0;
+	int pixels_to_draw = 256;
+	if (m_cont_reg[27] & 0x07)
+	{
+		// Scroll max 7 pixels to the right
+		int dot_scroll = m_cont_reg[27] & 0x07;
```

Comment:
Think you can do this assignment outside the if block, i.e.
```
int dot_scroll = m_cont_reg[27] & 0x07;
if (dot_scroll)
{
	scroll_to_mask = 8 - dot_scroll;
    // ...
}
```

PR otherwise looks good to me, assuming that this block isn't accessible by regular V9938 at all.